File: frontend/lib/features/shopping/data/dto/shopping_item.dart
class ShoppingItem {
  const ShoppingItem({
    required this.id,
    required this.title,
    required this.checked,
    required this.createdBy,
  });

  factory ShoppingItem.fromJson(Map<String, dynamic> j) => ShoppingItem(
        id: j['id'] as String? ?? '',
        title: j['title'] as String? ?? '',
        checked: j['checked'] as bool? ?? false,
        createdBy: j['createdBy'] as String? ?? '',
      );

  final String id;
  final String title;
  final bool checked;
  final String createdBy;
}

File: backend/src/main/java/com/aifb/platform/shopping/domain/ShoppingItem.java
package com.aifb.platform.shopping.domain;

import com.aifb.platform.common.persistence.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.util.UUID;

@Entity
@Table(name = "shopping_items")
public class ShoppingItem extends BaseEntity {

    @Column(name = "household_id", nullable = false)
    private UUID householdId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false)
    private boolean checked = false;

    @Column(name = "created_by", nullable = false)
    private UUID createdBy;

    protected ShoppingItem() {
    }

    public ShoppingItem(UUID householdId, String title, UUID createdBy) {
        this.id = UUID.randomUUID();
        this.householdId = householdId;
        this.title = title;
        this.createdBy = createdBy;
        this.checked = false;
    }

    public UUID getHouseholdId() { return householdId; }
    public String getTitle() { return title; }
    public boolean isChecked() { return checked; }
    public UUID getCreatedBy() { return createdBy; }

    public void toggle() {
        this.checked = !this.checked;
    }
}
